plugins/module_utils/network/sonic/config/qos_wred/qos_wred.py

```diff
@@ -306,52 +307,42 @@ def get_delete_qos_wred_requests(self, commands, have, is_delete_all):
-                    if green:
-                        enable = green.get('enable')
-                        min_threshold = green.get('min_threshold')
-                        max_threshold = green.get('max_threshold')
-                        drop_probability = green.get('drop_probability')
-
-                        if cfg_green:
-                            green_dict = {}
-                            cfg_enable = cfg_green.get('enable')
-                            cfg_min_threshold = cfg_green.get('min_threshold')
-                            cfg_max_threshold = cfg_green.get('max_threshold')
-                            cfg_drop_probability = cfg_green.get('drop_probability')
-
-                            if enable is not None and enable == cfg_enable:
-                                requests.append(self.get_delete_wred_cfg_attr(name, 'wred-green-enable'))
-                                green_dict['enable'] = enable
-                            if min_threshold and min_threshold == cfg_min_threshold:
-                                requests.append(self.get_delete_wred_cfg_attr(name, 'green-min-threshold'))
-                                green_dict['min_threshold'] = min_threshold
-                            if max_threshold and max_threshold == cfg_max_threshold:
-                                requests.append(self.get_delete_wred_cfg_attr(name, 'green-max-threshold'))
-                                green_dict['max_threshold'] = max_threshold
-                            if drop_probability is not None and drop_probability == cfg_drop_probability:
-                                requests.append(self.get_delete_wred_cfg_attr(name, 'green-drop-probability'))
-                                green_dict['drop_probability'] = drop_probability
-                            if green_dict:
-                                wred_dict.update({'name': name, 'green': green_dict})
-                    # Deletion my WRED profile name
-                    if not ecn and not green:
+                    for color in colors:
+                        color_dict = wred.get(color)
+                        cfg_color_dict = cfg_wred.get(color)
+                        color_attr_dict = {}
+                        if color_dict and cfg_color_dict:
+                            for attr in attributes:
+                                val = color_dict.get(attr)
+                                cfg_val = cfg_color_dict.get(attr)
+                                if attr == 'enable':
+                                    if val is not None and val == cfg_val:
+                                        requests.append(self.get_delete_wred_cfg_attr(name, f'wred-{color}-{attr}'))
+                                else:
+                                    if val and val == cfg_val:
+                                        requests.append(self.get_delete_wred_cfg_attr(name, f'{color}-{attr.replace("_", "-")}'))
```

Comment:
Since 'drop_probabliity' can be zero, perform a 'None' check
```suggestion
                                if val is not None and val == cfg_val:
                                    if attr == 'enable':
                                        requests.append(self.get_delete_wred_cfg_attr(name, f'wred-{color}-{attr}'))
                                    else:
                                        requests.append(self.get_delete_wred_cfg_attr(name, f'{color}-{attr.replace("_", "-")}'))
```